### PR TITLE
fix(query): fix binder create table

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

1.  fix binder create table with lower case column name
2. add `TEXT`, `CHARACTER` token

Fixes #issue

### CODE REVIEW

File: common/ast/src/parser/token.rs [hunk 1]
```diff
@@ -293,6 +293,8 @@ pub enum TokenKind {
     CONNECTION,
     #[token("CHAR", ignore(ascii_case))]
     CHAR,
+    #[token("CHARACTER", ignore(ascii_case))]
```
Comment (xudong963) [inline common/ast/src/parser/token.rs:296]:
```
    // Keywords
    //
    // Steps to add keyword:
    // 1. Add the keyword to token kind variants by alphabetical order.
    // 2. Search in this file to see if the new keyword is a commented
    //    out reserverd keyword. If so, uncomment the keyword in the
    //    reserved list.
```

**Review by xudong963 — COMMENTED** [review]
Parser-related golden test results need to update.

**Review by xudong963 — APPROVED** [review]

**Review by leiysky — APPROVED** [review]

**Comment by Xuanwo** [comment]
```shell
Following failed statements:
---------------------------------------------
Failed to execute. Collected info: 1105 (HY000): Code: 1005, displayText = error: 
  --> SQL:1:35
  |
1 | create table db1.t as SELECT CHAR(77,121,83,81,76) AS hello;
  |                                   ^^ expected `(`, `UNION`, `EXCEPT`, `INTERSECT`, or `SELECT`

.

Parsed Statement
    at_line: 61,
    s_type: Statement: ok, type: None,
    suite_name: gen/02_function/02_0043_function_char,
    text:
        create table db1.t as SELECT CHAR(77,121,83,81,76) AS hello;
    results: [],
    runs_on: {'mysql', 'clickhouse', 'http'},
---------------------------------------------
Expected: statement ok must get success response, not error code in response
Message: 
Parsed Statement
    at_line: 61,
    s_type: Statement: ok, type: None,
    suite_name: gen/02_function/02_0043_function_char,
    text:
        create table db1.t as SELECT CHAR(77,121,83,81,76) AS hello;
    results: [],
    runs_on: {'mysql', 'clickhouse', 'http'},
---------------------------------------------
Failed to execute. Collected info: Orig exception: Code: 1005, displayText = error: 
  --> SQL:1:35
  |
1 | create table db1.t as SELECT CHAR(77,121,83,81,76) AS hello;
  |                                   ^^ expected `(`, `UNION`, `EXCEPT`, `INTERSECT`, or `SELECT`

.

Parsed Statement
    at_line: 61,
    s_type: Statement: ok, type: None,
    suite_name: gen/02_function/02_0043_function_char,
    text:
        create table db1.t as SELECT CHAR(77,121,83,81,76) AS hello;
    results: [],
    runs_on: {'mysql', 'clickhouse', 'http'},
```

**Comment by Xuanwo** [comment]
@mergify update

**Pull request merged**